app/views/publish/courses/engineers_teach_physics/_edit_form_fields.html.erb

```diff
@@ -41,6 +41,11 @@
         <%= form.hidden_field(:subjects_ids, multiple: true, value: field_value) %>
       <% end %>
     <% end %>
+    <% if params[:publish_engineers_teach_physics_form]&.dig(:subjects_ids) %>
+      <% params[:publish_engineers_teach_physics_form][:subjects_ids].each do |field_value| %>
+        <%= form.hidden_field(:subjects_ids, multiple: true, value: field_value) %>
+      <% end %>
+    <% end %>
```

Comment:
```suggestion
    <% params.dig(:publish_engineers_teach_physics_form, :subjects_ids)&.each do |field_value| %>
        <%= form.hidden_field(:subjects_ids, multiple: true, value: field_value) %>
    <% end %>
```

Reply:
Much better

Reply:
we have spec/features/publish/courses/editing_engineers_teach_physics_spec.rb

Reply:
Updated spec